Bloom.md

```diff
@@ -0,0 +1,604 @@
+
+```
+127.0.0.1:6379> BF.INFO bloomobject1
+ 1) Capacity
+ 2) (integer) 1000
+ 3) Size
+ 4) (integer) 1431
+ 5) Number of filters
+ 6) (integer) 1
+ 7) Number of items inserted
+ 8) (integer) 1
+ 9) Expansion rate
+10) (integer) 2
+127.0.0.1:6379> BF.INFO bloomobject1 SIZE
+1) (integer) 1431
+127.0.0.1:6379> BF.INFO bloomobject1 CAPACITY
+1) (integer) 1000
+127.0.0.1:6379> BF.INFO bloomobject1 FILTERS
+1) (integer) 1
+```
+
+**`BF.RESERVE <key> <false_positive_rate> <capacity> [EXPANSION <expansion>] | [NONSCALING]`**
+
+This API is used to create a bloom object with specific properties.
+When the command is used, only either EXPANSION or NONSCALING can be used. If both are used, an error is returned.
+
+"CAPACITY" is the number of items that the users wants to store on the bloom object. For non scaling, this is a limit on
+the number of items that can be inserted. For scaling, this is the number of items that can be added after which scaling
+occurs.
+
+"EXPANSION" can be used to create a scaling enabled bloom object with the specifed expansion rate.
+
+"NONSCALING" can be used to indicate that the bloom object should not auto scale once items are added such that it reaches
+full capacity.
+
+The response is a simple String reply with OK indicating successful creation.
+```
+OK
+```
+
+**`BF.INSERT <key> [CAPACITY <capacity>] [ERROR <fp_error>] [EXPANSION <expansion>] [NOCREATE] [NONSCALING] ITEMS <item> [<item> ...]`**
+
+This API is used to create a bloom object with specific properties and add item/s to it.
+
+"CAPACITY" is the number of items that the users wants to store on the bloom object. For non scaling, this is a limit on
+the number of items that can be inserted. For scaling, this is the number of items that can be added after which scaling
+occurs.
+
+"ERROR" is the false positive error rate.
+
+"NOCREATE" can be used to specify that the command should not result in creation of a new bloom object if it does not exist.
+If NOCREATE is used along with CAPACITY or ERROR, an error is returned.
+
+"EXPANSION" can be used to create a scaling enabled bloom object with the specifed expansion rate.
+
+"NONSCALING" can be used to indicate that the bloom object should not auto scale once items are added such that it reaches
+full capacity. Only either EXPANSION or NONSCALING can be used. If both are used, an error is returned.
+
+"ITEMS" can be used to list one or more items to add to the bloom object.
+
+The response is an array reply with one or more Integer replies.
+1 indicates the item does not exist (based on false positve rate) yet and was added successfully to the bloom filter.
+If a bloom object named <key> does not exist, the bloom object is created and the item will be added to it.
+0 indicates the item already exists (based on false positve rate).
+```
+(integer) 1
+(integer) 1
+(integer) 0
+```
+
+The following are NEW commands which are not included in ReBloom:
+
+**`BF.LOAD <key> <serialized-value-dump> <TTL>`**
```

Comment:
TTL will not be provided in the command since this will be added to the AOF automatically on the items which have a TTL set. I will update the RFC